pkg/neorpc/filters.go

```diff
@@ -49,6 +54,16 @@ type (
 	}
 )
 
+// SubscriptionFilter is an interface for all subscription filters.
+type SubscriptionFilter interface {
+	// IsValid checks whether the filter is valid and returns
+	// a specific [ErrInvalidSubscriptionFilter] error if not.
+	IsValid() error
+}
+
+// ErrInvalidSubscriptionFilter is returned when the filter is invalid.
```

Comment:
s/the filter/the subscription filter

Just because we're at `neorpc` package, so from the user's POW it would be `neorpc.ErrInvalidSubscriptionFilter`, and documentation should be a little more precise.